src/components/token/NewUiTokenAutoCompleteField.js

```diff
@@ -0,0 +1,245 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import tokensApi from '../../api/tokensApi';
+import { debounce, get } from 'lodash';
+import { constants as hathorLibConstants } from '@hathor/wallet-lib';
+import { connect } from 'react-redux';
+
+const DEBOUNCE_SEARCH_TIME = 200; // ms
+
+const mapStateToProps = state => {
+  return { serverInfo: state.serverInfo };
+};
+
+class NewUiTokenAutoCompleteField extends React.Component {
```

Comment:
question: This duplication of code raises some worries here, because we will soon need to refactor those components to functional. This unification of the logic between files - even though it's supposed to be the same logic - can generate accidental bugs.

Would it be possible for us to find another way of making this distinction between the new UI and the old UI without using hooks? That way, we could have the same `renderUi()` / `renderNewUi()` pattern implemented in those components.

A first possibility that occurs to me is to create a new Redux property storing this new UI information and use it in the legacy class components. That property could be updated by the hook itself, since it's running correctly at the higher level components.

Do you think this is a valid solution, or maybe would you have other ideas on how to implement an alternative?

Reply:
I share the concern about potential issues with code duplication, especially considering future refactoring to functional components. Here’s an approach we could take to simplify this while keeping the same `renderUi` and `renderNewUi` pattern:

- We can maintain a single file for the class components.
- The higher-level component, which is functional and has access to the feature flag, could pass down a prop to the class component, indicating whether to render the new or old UI.
- This way, the class component can determine the appropriate UI to render without needing hooks directly, preserving a single source of truth and reducing the risk of accidental bugs due to duplicated logic.

Let me know if this approach seems feasible or if you have any additional thoughts!

Reply:
That looks like a great solution to me :+1: . We can move on with this approach.

Reply:
Just a quick note: since some components rely on inheritance, I wasn't able to create new methods specifically for the new UI. Instead, I adjusted the content conditionally based on the feature flag.